In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

**1. Есть две таблички. В первой – юзеры, пришедшие в феврале с Apple Search Ads, и их покупки. Во второй – косты по Apple Search Ads в разрезе кампаний.** 

Задача: 
- Найти, как минимум, две кампании, в рамках которых происходило что-то странное, и попробовать объяснить причину. 
- Найти три наиболее прибыльные кампании и три наиболее убыточные.
- В разрезе US кампании проанализировать пользователей и найти какие-либо закономерности.


In [2]:
df = pd.read_csv('/mnt/HC_Volume_18315164/home-jupyter/jupyter-artem-kuznetsov/costs_asa_campaign_filtered.csv')
df.head(10)

Date    Campaign   Cost
0  2023-02-01  US Common_  89,22
1  2023-02-01  NZ Common_  21,54
2  2023-02-01  DE Common_  18,15
3  2023-02-01  UK Common_  11,21
4  2023-02-01  FR Common_   7,83
5  2023-02-01    AU SKAG_   7,24
6  2023-02-01  CA Common_   6,02
7  2023-02-01  JP Common_   5,15
8  2023-02-01     GR SKAG   5,05
9  2023-02-01  NL Common_   4,10

Посмотрим на типы данных в таблице


In [3]:
df.dtypes

Date        object
Campaign    object
Cost        object
dtype: object

Преобразуем колонку Cost в числовой формат

In [4]:
df['Cost'] = df['Cost'].str.replace(',', '.') # поменяем разделить в колонке

In [5]:
df['Cost'] = df['Cost'].astype(float)

In [6]:
df.dtypes

Date         object
Campaign     object
Cost        float64
dtype: object

Найдем три наиболее прибыльные и наиболее убыточные компании

In [7]:
max_profit = df.groupby('Campaign')\
    .agg({'Cost':'sum'})\
    .reset_index().sort_values('Cost', ascending=False)
max_profit

Campaign     Cost
100  US Common_  3487.89
67   NZ Common_  1367.59
19   DE Common_   417.11
29   FR Common_   406.30
98   UK Common_   406.24
..          ...      ...
24      EC SKAG     0.86
76      PK SKAG     0.67
51      KH SKAG     0.59
9       AZ SKAG     0.44
60      MO SKAG     0.16

[107 rows x 2 columns]

**Максимально прибыльные компании:** 
   
   **1. US Common_** 
    
   **2. NZ Common_**
    
   **3. DE Common_**
    

In [8]:
min_profit = df.groupby('Campaign')\
    .agg({'Cost':'sum'})\
    .reset_index().sort_values('Cost', ascending=True)
min_profit

Campaign     Cost
60      MO SKAG     0.16
9       AZ SKAG     0.44
51      KH SKAG     0.59
76      PK SKAG     0.67
24      EC SKAG     0.86
..          ...      ...
98   UK Common_   406.24
29   FR Common_   406.30
19   DE Common_   417.11
67   NZ Common_  1367.59
100  US Common_  3487.89

[107 rows x 2 columns]

**Максимально убыточные компании:** 
   
   **1. MO SKAG** 
    
   **2. AZ SKAG**
    
   **3. KH SKAG**

______________________________________________________________________________________________________

In [9]:
df_1 = pd.read_csv('/mnt/HC_Volume_18315164/home-jupyter/jupyter-artem-kuznetsov/users_asa_feb_flitered.csv')
df_1.head(10)

user_id   lang  pro  \
0  FA64312F-273A-41AB-BB9C-DB43AFE5B760  de_PL    0   
1  3FF6E426-B5D5-4C8B-A915-9FB58464EA50  en_NZ    0   
2  9C82BE9F-0212-47EA-872F-AF7237D43064  en_CA    1   
3  750657AF-3BB4-457A-9BBA-BE691CA5A75C  en_AR    0   
4  A417C7DE-229E-4BB3-8BF0-17FA6883535E  ru_UA    0   
5  47D57894-8B0B-441F-A3CB-D35126C0DCCF  en_NZ    0   
6  71F5F711-36E2-4BAB-8515-011A8DC39201  en_ZA    1   
7  1C597600-41BA-4CC9-B9F3-E6FC31034C1A  it_IT    0   
8  19C3747E-4396-41E8-ADCC-4BAEA4565728  en_AU    0   
9  743235CF-46B5-4C8E-AC57-30D0C70EC275  de_DE    0   

                               product  price  sail  kite  wind  surf  fish  \
0                                  NaN    NaN     0     0     0     0     0   
1                                  NaN    NaN     1     0     0     0     1   
2  co.windyapp.pro_sub_year_50_notrial   50.0     1     0     0     0     1   
3                                  NaN    NaN     0     1     0     0     0   
4                                  NaN    NaN     0     0     0     0     0   
5                                  NaN    NaN     0     0     0     0     0   
6  co.windyapp.pro_sub_year_50_trial7d   50.0     0     0     0     0     0   
7                                  NaN    NaN     0     0     0     0     1   
8                                  NaN    NaN     0     0     0     1     0   
9                                  NaN    NaN     0     0     0     0     0   

   snow  other            source      campaign install_date  
0     0      1  Apple Search Ads  PL Discovery   01.02.2023  
1     0      0  Apple Search Ads    NZ Common_   01.02.2023  
2     0      0  Apple Search Ads    CA Common_   01.02.2023  
3     0      0  Apple Search Ads       AR Skag   01.02.2023  
4     0      1  Apple Search Ads       UA SKAG   01.02.2023  
5     0      0  Apple Search Ads    NZ Common_   01.02.2023  
6     0      1  Apple Search Ads  ZA Discovery   01.02.2023  
7     0      0  Apple Search Ads    IT Common_   01.02.2023  
8     0      0  Apple Search Ads  AU Discovery   01.02.2023  
9     0      0  Apple Search Ads    DE Common_   01.02.2023

In [10]:
US = df_1.query("campaign=='US Common_'")
US.head(10)

user_id   lang  pro  \
10  87AF7E44-0DFF-4912-B44D-4F9A32E4A7C5  en_US    1   
11  E2ABD56C-E797-4815-898F-949D514A894A  en_US    0   
13  63909BE8-514B-41AE-93BD-2ABA79B7E15F  fr_FR    0   
19  05195C89-79B6-4CF5-911F-96A366C89DE2  en_US    0   
51  D77E108C-70A5-4865-A65F-87811E1D131D  en_US    0   
58  FB142B07-6A13-45CB-B893-3547C0858CC0  en_US    0   
59  1CFA4CF8-8149-49EC-B0EE-9E01AF516C8D  en_US    0   
65  820AC31F-BBBF-4F6C-90AD-D96E350CFC97    NaN    0   
67  B469A3FF-BC13-4629-87BE-EA30D6D0B918  en_CA    1   
71  D5844F47-8710-4F60-92F1-A57532BFFD2E  en_US    0   

                                      product  price  sail  kite  wind  surf  \
10  co.windyapp.2210.pro_sub_year_100_trial7d  100.0     0     0     0     0   
11                                        NaN    NaN     0     0     0     0   
13                                        NaN    NaN     0     1     0     0   
19                                        NaN    NaN     0     0     0     0   
51                                        NaN    NaN     0     0     0     0   
58                                        NaN    NaN     0     0     0     0   
59                                        NaN    NaN     0     0     0     1   
65                                        NaN    NaN     0     0     0     0   
67                co.windyapp.pro_forever_150  150.0     1     0     0     0   
71                                        NaN    NaN     0     0     0     0   

    fish  snow  other            source    campaign install_date  
10     1     0      0  Apple Search Ads  US Common_   01.02.2023  
11     0     0      1  Apple Search Ads  US Common_   01.02.2023  
13     0     0      0  Apple Search Ads  US Common_   01.02.2023  
19     1     0      0  Apple Search Ads  US Common_   01.02.2023  
51     0     0      0  Apple Search Ads  US Common_   01.02.2023  
58     0     0      0  Apple Search Ads  US Common_   01.02.2023  
59     0     0      0  Apple Search Ads  US Common_   01.02.2023  
65     0     0      0  Apple Search Ads  US Common_   01.02.2023  
67     0     0      0  Apple Search Ads  US Common_   01.02.2023  
71     0     1      1  Apple Search Ads  US Common_   01.02.2023

In [11]:
#заменим Nan значени на 0
US = US.fillna(0)
US.head(10)

user_id   lang  pro  \
10  87AF7E44-0DFF-4912-B44D-4F9A32E4A7C5  en_US    1   
11  E2ABD56C-E797-4815-898F-949D514A894A  en_US    0   
13  63909BE8-514B-41AE-93BD-2ABA79B7E15F  fr_FR    0   
19  05195C89-79B6-4CF5-911F-96A366C89DE2  en_US    0   
51  D77E108C-70A5-4865-A65F-87811E1D131D  en_US    0   
58  FB142B07-6A13-45CB-B893-3547C0858CC0  en_US    0   
59  1CFA4CF8-8149-49EC-B0EE-9E01AF516C8D  en_US    0   
65  820AC31F-BBBF-4F6C-90AD-D96E350CFC97      0    0   
67  B469A3FF-BC13-4629-87BE-EA30D6D0B918  en_CA    1   
71  D5844F47-8710-4F60-92F1-A57532BFFD2E  en_US    0   

                                      product  price  sail  kite  wind  surf  \
10  co.windyapp.2210.pro_sub_year_100_trial7d  100.0     0     0     0     0   
11                                          0    0.0     0     0     0     0   
13                                          0    0.0     0     1     0     0   
19                                          0    0.0     0     0     0     0   
51                                          0    0.0     0     0     0     0   
58                                          0    0.0     0     0     0     0   
59                                          0    0.0     0     0     0     1   
65                                          0    0.0     0     0     0     0   
67                co.windyapp.pro_forever_150  150.0     1     0     0     0   
71                                          0    0.0     0     0     0     0   

    fish  snow  other            source    campaign install_date  
10     1     0      0  Apple Search Ads  US Common_   01.02.2023  
11     0     0      1  Apple Search Ads  US Common_   01.02.2023  
13     0     0      0  Apple Search Ads  US Common_   01.02.2023  
19     1     0      0  Apple Search Ads  US Common_   01.02.2023  
51     0     0      0  Apple Search Ads  US Common_   01.02.2023  
58     0     0      0  Apple Search Ads  US Common_   01.02.2023  
59     0     0      0  Apple Search Ads  US Common_   01.02.2023  
65     0     0      0  Apple Search Ads  US Common_   01.02.2023  
67     0     0      0  Apple Search Ads  US Common_   01.02.2023  
71     0     1      1  Apple Search Ads  US Common_   01.02.2023

In [12]:
# удалим дубликаты

In [13]:
US.drop_duplicates()

user_id   lang  pro  \
10    87AF7E44-0DFF-4912-B44D-4F9A32E4A7C5  en_US    1   
11    E2ABD56C-E797-4815-898F-949D514A894A  en_US    0   
13    63909BE8-514B-41AE-93BD-2ABA79B7E15F  fr_FR    0   
19    05195C89-79B6-4CF5-911F-96A366C89DE2  en_US    0   
51    D77E108C-70A5-4865-A65F-87811E1D131D  en_US    0   
...                                    ...    ...  ...   
6289  89669F25-2124-4C88-9179-F5481605C8F0  en_US    0   
6295  243D3A2C-291B-41E7-A7EC-905F57129F43  en_US    0   
6298  4D2F4D59-B947-49C8-919E-BFC97320DD2F  en_US    0   
6299  F9A08623-46F9-4E06-BB1A-09A945FB64DF  en_US    0   
6306  8EBD6D6A-BAB4-4878-ADE1-5EB92BA26E3C  en_US    0   

                                        product  price  sail  kite  wind  \
10    co.windyapp.2210.pro_sub_year_100_trial7d  100.0     0     0     0   
11                                            0    0.0     0     0     0   
13                                            0    0.0     0     1     0   
19                                            0    0.0     0     0     0   
51                                            0    0.0     0     0     0   
...                                         ...    ...   ...   ...   ...   
6289                                          0    0.0     0     0     0   
6295                                          0    0.0     0     0     0   
6298                                          0    0.0     1     0     0   
6299                                          0    0.0     0     0     0   
6306                                          0    0.0     0     0     0   

      surf  fish  snow  other            source    campaign install_date  
10       0     1     0      0  Apple Search Ads  US Common_   01.02.2023  
11       0     0     0      1  Apple Search Ads  US Common_   01.02.2023  
13       0     0     0      0  Apple Search Ads  US Common_   01.02.2023  
19       0     1     0      0  Apple Search Ads  US Common_   01.02.2023  
51       0     0     0      0  Apple Search Ads  US Common_   01.02.2023  
...    ...   ...   ...    ...               ...         ...          ...  
6289     0     0     0      0  Apple Search Ads  US Common_   28.02.2023  
6295     0     0     0      1  Apple Search Ads  US Common_   28.02.2023  
6298     0     0     0      0  Apple Search Ads  US Common_   28.02.2023  
6299     0     1     0      0  Apple Search Ads  US Common_   28.02.2023  
6306     0     1     0      0  Apple Search Ads  US Common_   28.02.2023  

[1219 rows x 15 columns]

In [14]:
# количество подписок в каждом сегменте
US[['pro','price','sail','kite','wind','surf','fish','snow']].sum ()

pro        93.0
price    8110.0
sail      229.0
kite       29.0
wind       28.0
surf      102.0
fish      308.0
snow       74.0
dtype: float64

In [15]:
US['product'].unique()

array(['co.windyapp.2210.pro_sub_year_100_trial7d', 0,
       'co.windyapp.pro_forever_150',
       'co.windyapp.pro_sub_year_60_trial7d',
       'co.windyapp.2210.pro_sub_year_80_notrial',
       'co.windyapp.pro_sub_year_30_notrial',
       'co.windyapp.2110.pro_sub_year_40_notrial',
       'co.windyapp.2110.pro_sub_year_60_trial7d',
       'co.windyapp.2110.pro_sub_month_10_notrial',
       'co.windyapp.pro_sub_year_50_trial7d',
       'co.windyapp.pro_forever_180',
       'co.windyapp.pro_sub_year_35_notrial',
       'co.windyapp.2110.pro_sub_year_60_notrial'], dtype=object)

**93 пользователя оформило pro подписку. Чаще всего пользователи пользуются приложением для рыбалки. Реже всего для кайта и виндсерфинга. Для серфинга на втором месте.**

_____________________________________________________________

**2. Наиболее проблемным каналом трафика для нас является Facebook. Мы тестируем большое количество креативов каждый месяц. При этом большая часть из этих креативов показывает хорошие показатели в других каналах трафика, но не в Фейсбуке.**

**Задача:**
- Посмотреть четыре представленных ниже креатива и сделать предположение, какой из них оказался наиболее эффективным. 
- Предположить, что можно было бы в каждом из креативов поменять, чтобы существенно улучшить одну из трех ключевых конверсий (конверсия в клик, конверсия в инсталл, конверсия в старт пробной версии приложения).


**Думаю, что наиболее эффективным оказался креатив Fish @ Solunar**

**Креативе bubblefish сделать больше акцент на пойманной крупной рыбе. Сделать этот момент ярче.**

**Креатив fishing calm нет ничего, что может зацепить. Не понятно как работает и для чего вообще нужно приложение**

**Креатив  windy fish в целом неплохой, но кажется, что усложнен в конце**

___________________________________________________________________

**3. Почти 55% ревенью (за год) мы получаем от юзеров в первую неделю их жизни. Оставшиеся 45% получаем в течение года в первую очередь за счет распродаж (проводим их 1-2 раза в месяц, делаем скидки до 70%).** 

**Задача:**
- Предложить механики внутри приложения или за его пределами, которые позволили бы более эффективно конвертить в покупку юзеров, ничего не купивших в первую неделю.


**- Предоставить пользователям которые не сделали покупку в первую неделю персональные скидки, ограниченные по времени**

**- Поощрять пользователей которые сделали покупку в первую неделю, бонусами, промокодами и т.д.**

**- Выводить на главный экран кнопку "купить"**

**- Создать иллюзию срочности покупки**